examples/isometric/game.go

```diff
@@ -0,0 +1,223 @@
+// Copyright 2021 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build example
+// +build example
+
+package main
+
+import (
+	"fmt"
+	"image"
+
+	"github.com/hajimehoshi/ebiten/v2"
+	"github.com/hajimehoshi/ebiten/v2/ebitenutil"
+	"github.com/hajimehoshi/ebiten/v2/inpututil"
+)
+
+var spinner = []byte(`-\|/`)
+
+// Game is an isometric demo game.
+type Game struct {
+	w, h         int
+	currentLevel *Level
+
+	camX, camY float64
+	camScale   float64
+	camScaleTo float64
+
+	mousePanX, mousePanY int
+
+	spinnerIndex int
+
+	op *ebiten.DrawImageOptions
+}
+
+// NewGame returns a new isometric demo Game.
+func NewGame() *Game {
+	return &Game{
+		currentLevel: NewLevel(),
+		camScale:     2.0,
+		camScaleTo:   2.0,
+		op: &ebiten.DrawImageOptions{
+			Filter: ebiten.FilterNearest,
+		},
+	}
+}
+
+// Update reads current user input and updates the Game state.
+func (g *Game) Update() error {
+	// Update target zoom level.
+	var scrollY float64
+	if inpututil.KeyPressDuration(ebiten.KeyC) > 0 || inpututil.KeyPressDuration(ebiten.KeyPageDown) > 0 {
+		scrollY = -0.15
+	} else if inpututil.KeyPressDuration(ebiten.KeyE) > 0 || inpututil.KeyPressDuration(ebiten.KeyPageUp) > 0 {
+		scrollY = .15
+	} else {
+		_, scrollY = ebiten.Wheel()
+		if scrollY < -1 {
+			scrollY = -1
+		} else if scrollY > 1 {
+			scrollY = 1
+		}
+	}
+	g.camScaleTo += scrollY * (g.camScaleTo / 7)
+
+	// Clamp target zoom level.
+	if g.camScaleTo < 0.01 {
+		g.camScaleTo = 0.01
+	} else if g.camScaleTo > 100 {
+		g.camScaleTo = 100
+	}
+
+	// Smooth zoom transition.
+	div := 10.0
+	if g.camScaleTo > g.camScale {
+		g.camScale += (g.camScaleTo - g.camScale) / div
+	} else if g.camScaleTo < g.camScale {
+		g.camScale -= (g.camScale - g.camScaleTo) / div
+	}
+
+	pan := 4.0 / g.camScale
+	// Pan camera via keyboard.
+	if inpututil.KeyPressDuration(ebiten.KeyLeft) > 0 || inpututil.KeyPressDuration(ebiten.KeyA) > 0 {
```

Comment:
Use `ebiten.IsKeyPressed(key)` instead of `inpututil.KeyPressDurations(key) > 0`.